core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -811,7 +872,7 @@ object UnitActions {
         if (unit.isFortified() || unit.canFortify() || unit.currentMovement == 0f) return
         // If this unit is working on an improvement, it cannot sleep
         if (unit.currentTile.hasImprovementInProgress()
-            && unit.canBuildImprovement(unit.currentTile.getTileImprovementInProgress()!!)) return
+            && (unit.currentTile.improvementInProgress == Constants.repair || unit.canBuildImprovement(unit.currentTile.getTileImprovementInProgress()!!))) return
```

Comment:
I think that unit.canBuildImprovement should return 'true' for "repair", and then we don't need this "if/else" here

Reply:
Already does, so can remove the extra OR here